ldap/ldap_test.go

```diff
@@ -0,0 +1,46 @@
+package ldap
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLDAPConstructorRequiresNonEmptyArgs(t *testing.T) {
+	assert := assert.New(t)
+
+	l, err := NewLDAPAuthenticator("foo", "bar", "baz")
+	assert.NotNil(l)
+	assert.NoError(err)
+
+	l, err = NewLDAPAuthenticator("", "bar", "baz")
+	assert.Nil(l)
+	assert.Error(err)
+
+	l, err = NewLDAPAuthenticator("foo", "", "baz")
+	assert.Nil(l)
+	assert.Error(err)
+
+	l, err = NewLDAPAuthenticator("foo", "bar", "")
+	assert.Nil(l)
+	assert.Error(err)
+}
+
+// This test requires an LDAP server. Uncomment to test.
```

Comment:
should we make this an interface so we can mock out the stuff that's built upon it?